### DISCUSSION TITLE
Difference between the following movers PackRotamersMoverPartGreedy and FastRelax/FastDesign.

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#198] [Rosetta] [answered]
How does the Greedy mover compare to FastRelax/Design for sidechain packing? If I want to optimize how tightly packed sidechains are should I use the Greedy Mover over iterations of FastRelax/Design?

**Comment by roccomoretti** [comment, marked as answer]
[PackRotamersMoverPartGreedy](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/movers_pages/PackRotamersMoverPartGreedyMover) allows you to specify a set of target residues and then do an exhaustive residue-by-residue sample of possible sidechain identities/conformations for residues nearby the target residues. It does include a minimization step, but only for those residues near the target residues. All other residues are handled exactly as they would be for the regular PackRotamersMover. 

It's really an extension of the PackRotamersMover, with an extra optimization step for residues around the target residues. In particular, it's a fixed-backbone protocol -- only the sidechain conformations/identities will change. The backbone won't move. In contrast, FastRelax/FastDesign is a protocol intended to allow the backbone to move during the optimization.

If you want backbone optimization, then you'll want the FastRelax/FastDesign mover. If you don't have a target residue list, around which you want to do optimization, then there's not really any point in PackRotamersMoverPartGreedy -- just use the regular PackRotamersMover, or if you want to include sidechain (but not backbone) minimization, use the MinPacker.

Possibly helpful info for determining use cases: The PackRotamersMoverPartGreedy comes out of the enzyme design effort. The usual "target residues" are the transition state & catalytic residues (a.k.a. "theozyme") which enzyme design is based around. These need to be well-backed, hence the mover which attempts to do more extensive sampling around these particular residues. 

I'll also note that these days it might be better to take advantage of the new(er) ResidueSelector logic to define your shells (target residues, neighbors, everything else) and then apply the MinPacker over the neighbor selector and the regular PackRotamersMover over the "everything else" one. That should give you more control over sampling with potentially better results.

**Reply by tsamdin** [reply]
Thank you! This answered my question.
